lib/pbench/test/unit/server/test_compare_datasets.py

```diff
@@ -0,0 +1,150 @@
+from http import HTTPStatus
+from pathlib import Path
+
+from pquisby.lib.post_processing import QuisbyProcessing
+import pytest
+import requests
+
+from pbench.server import JSON
+from pbench.server.cache_manager import CacheManager, Tarball
+from pbench.server.database.models.datasets import Dataset, DatasetNotFound, Metadata
+
+
+class TestCompareDatasets:
+    @pytest.fixture()
+    def query_get_as(self, client, server_config, more_datasets, get_token_func):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+            more_datasets: Dataset construction fixture
+            get_token_func: Pbench token fixture
+        """
+
+        def query_api(
+            datasets: list, user, expected_status: HTTPStatus
+        ) -> requests.Response:
+            ds_list = []
+            for dataset in datasets:
+                try:
+                    dataset_id = Dataset.query(name=dataset).resource_id
+                    ds_list.append(dataset_id)
+                except DatasetNotFound:
+                    ds_list.append(dataset)  # Allow passing deliberately bad value
+
+            headers = {"authorization": f"bearer {get_token_func(user)}"}
+            response = client.get(
+                f"{server_config.rest_uri}/compare",
+                query_string={"datasets": ds_list},
+                headers=headers,
+            )
+            assert response.status_code == expected_status
+            return response
+
+        return query_api
+
+    def mock_find_dataset(self, _dataset: str) -> Tarball:
+        class Tarball(object):
+            tarball_path = Path("/dataset/tarball.tar.xz")
+            name = "tarball"
+
+            def extract(_tarball_path: Path, _path: str) -> str:
+                return "CSV_file_as_a_byte_stream"
+
+        return Tarball
```

Comment:
This is returning the (mock) `Tarball` _class_...I think we are supposed to be returning a `Tarball` _instance_ here.  I.e., this should be:
```python
        return Tarball()
```

Ditto line 133.